File: math/code/stern-brocot/fraction-finding-3.cpp
#include <iostream>
#include <tuple>
#include <vector>

// --8<-- [start:core]
// Locate a given fraction in the Stern-Brocot tree.
auto find(int x, int y) {
  std::vector<std::pair<int, char>> res;
  bool right = true;
  while (y) {
    res.emplace_back(x / y, right ? 'R' : 'L');
    x %= y;
    std::swap(x, y);
    right ^= 1;
  }
  --res.back().first;
  return res;
}

// --8<-- [end:core]
int main() {
  int x, y;
  std::cin >> x >> y;
  for (auto cr : find(x, y)) {
    for (; cr.first; --cr.first) {
      std::cout << cr.second;
    }
  }
  return 0;
}

File: math/code/stern-brocot/fraction-finding-3.py
# STERN-BROCOT TREE FRACTION FINDING.
# --8<-- [start:core]
# Locate a given fraction in the Stern-Brocot tree.
def find(x, y):
    res = []
    right = True
    while y:
        res.append([x // y, ("R" if right else "L")])
        x, y = y, x % y
        right ^= 1
    res[-1][0] -= 1
    return res


# --8<-- [end:core]
if __name__ == "__main__":
    x, y = map(int, input().split())
    print("".join(ch * t for t, ch in find(x, y)))
